### PR TITLE
windows: cli install

### PR DESCRIPTION
**Pull request by zaucy**
~~Waiting for #8791 to merge~~

Release Notes:

- Added windows suport for cli install command

### CODE REVIEW

**Comment by mikayla-maki** [comment]
Merged :)

File: crates/gpui/src/platform/windows/platform.rs [hunk 1]
```diff
@@ -279,7 +281,23 @@ impl Platform for WindowsPlatform {
 
     // todo!("windows")
```
Comment (mikayla-maki) [inline crates/gpui/src/platform/windows/platform.rs:282]: We should remove these if they've been implemented.

File: crates/install_cli/src/install_cli.rs [hunk 2]
```diff
@@ -5,6 +5,7 @@ use util::ResultExt;
 
 actions!(cli, [Install]);
 
+#[cfg(not(windows))]
```
Comment (mikayla-maki) [inline crates/install_cli/src/install_cli.rs:8]: Isn't this supposed to be `target_os = windows`?

File: crates/install_cli/src/install_cli.rs [hunk 3]
```diff
@@ -56,3 +57,39 @@ pub async fn install_cli(cx: &AsyncAppContext) -> Result<()> {
         Err(anyhow!("error running osascript"))
     }
 }
+
+#[cfg(windows)]
```
Comment (mikayla-maki) [inline crates/install_cli/src/install_cli.rs:61]: Same here as above.

File: crates/gpui/src/platform/windows/platform.rs [hunk 4]
```diff
@@ -279,7 +281,23 @@ impl Platform for WindowsPlatform {
...
     fn path_for_auxiliary_executable(&self, name: &str) -> Result<PathBuf> {
-        Err(anyhow!("not yet implemented"))
+        if name == "cli" {
```
Comment (mikayla-maki) [inline crates/gpui/src/platform/windows/platform.rs:284]: Why is this only being done for "cli"? Couldn't we just do this for everything?

File: crates/install_cli/src/install_cli.rs [hunk 5]
```diff
@@ -56,3 +57,39 @@ pub async fn install_cli(cx: &AsyncAppContext) -> Result<()> {
...
+    use winreg::enums::*;
+    use winreg::RegKey;
+
+    let cli_path = cx.update(|cx| cx.path_for_auxiliary_executable("cli"))??;
+    let cli_file_name = cli_path.file_name().expect("missing cli file name");
+    let cli_parent_path = cli_path.parent();
+    let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+    // https://learn.microsoft.com/en-us/windows/win32/shell/app-registration
+    let app_path_reg_path =
+        Path::new("Software\\Microsoft\\Windows\\CurrentVersion\\App Paths").join(cli_file_name);
```
Comment (mikayla-maki) [inline crates/install_cli/src/install_cli.rs:72]: Is this safe to hard code?

**Review by mikayla-maki — CHANGES_REQUESTED** [review]
Left a few small comments, but I'm wondering whether this should be implemented yet, I don't think we have a windows compatible CLI

File: crates/gpui/src/platform/windows/platform.rs (hunk 4 above)
Comment (zaucy) [reply crates/gpui/src/platform/windows/platform.rs:284]: Sorry was mostly just a draft. Really just airing out my thoughts.  I wasn't sure how to implement this function besides the cli condition.

Comment (mikayla-maki) [reply crates/gpui/src/platform/windows/platform.rs:284]: Gotcha, don't worry too much about the review then :)

File: crates/install_cli/src/install_cli.rs (hunk 5 above)
Comment (mikayla-maki) [reply crates/install_cli/src/install_cli.rs:72]: Perfect!

File: crates/install_cli/src/install_cli.rs (hunk 3 above)
Comment (mikayla-maki) [reply crates/install_cli/src/install_cli.rs:61]: Nevermind, I learned something about #[cfg] today!

File: crates/install_cli/src/install_cli.rs (hunk 2 above)
Comment (mikayla-maki) [reply crates/install_cli/src/install_cli.rs:8]: I am incorrect.

**Comment by mikayla-maki** [comment]
As this hasn't been touched in 2 weeks, closing.

**Pull request closed without merging**